### ISSUE TITLE
Accessibility Issues around Header Base component (HeaderBaseWActionsAndSwitcher) in Carbon

### ISSUE THREAD

**Issue by aliunwala** [carbon-design-system/carbon#6966]
This is the component I am referring to:
![image](https://user-images.githubusercontent.com/4451128/94983392-f8ae1400-0507-11eb-9e81-67e4321edd18.png)

The `HeaderBaseWActionsAndSwitcher` component has several accessibility issues that we have found. The issues mainly revolve around general usability when trying to access this component with a screen reader. Right now tabbing opens the menu but there is no indicator that this menu is closable etc. and then this menu and all sites get inserted into the tab order. 

A better solution is to make the `HeaderBaseWActionsAndSwitcher` a menu button. Normally this button would be tab focusable but then traversable by using the arrow keys. See the accessible code example here: https://www.w3.org/TR/wai-aria-practices-1.2/examples/menubar/menubar-1/menubar-1.html

This code example should be pretty close to what I understand this component should be. 😄 

Follow the aria authoring practices to construct this as a button with aria-haspopup=“true” and a proper role of “menu” and “menuitem” on the `<ul>` and `<li>` items beneath.

Additional notes that we saw from looking at the code:
1. You have this whole menu wrapped in a <nav> this is probably not necessary as I believe there is probably a nav higher up the DOM.

2. You are using `tabindex=“0”` on each of your `<li>` when they should be `tabindex=“-1”` and controlled using keypress events for the arrow keys. 

3. The menu is failing to handle dismissing of the menu on loss of focus. (Hopefully this will be partly resolved by removing the `<li>` items from the tab order)

4. Arrow based navigation should wrap around in the menu (Like the example [here](https://www.w3.org/TR/wai-aria-practices-1.2/examples/menubar/menubar-1/menubar-1.html) when you tab + use arrow keys)

5. The switch button should be given a meaningful name. Right now its default is `aria-label="Switch"` which is not very descriptive and causes confusion as the purpose of this menu. Something more like `aria-label="Switch between IBM Products"`

Other helpful links:
https://www.w3.org/TR/wai-aria-practices-1.2/#menu

## Environment

> Operating system
All
> Browser
All
> Automated testing tool and ruleset
n/a
> Assistive technology used to verify
You should be able to see the issues by just using tabbing, arrow keys, and the enter key. (Also shows up in voice over/JAWS)

**Comment by aliunwala** [comment]
Also I would like to note that I will be out of town next week and may be a bit slow to respond to any questions you might have on this one.

**Comment by tw15egan** [comment]
@vpicone is this part of the theme?

**Comment by vpicone** [comment]
@aliunwala you can (and should) have multiple navs in your page, I agree with your last point on a more descriptive label. If you'd like to open a PR for the others, you'd want to do it on the theme repo. @tw15egan yeah, I'll move this to the theme.
